File: 심수연/7주차/260210.py
# https://www.acmicpc.net/problem/7568

import sys
input = sys.stdin.readline

N = int(input())

arr = []

for i in range(N):
  arr.append(list(map(int, input().split())))

# [[55, 185], [58, 183], [88, 186], [60, 175], [46, 155]]

rank = [1 for _ in range(N)]

# 자기(j)보다 덩치 큰 사람 있으면 등수를 + 1씩 하면 됨
for i in range(N):
  for j in range(N):
    if arr[i][0] > arr[j][0] and arr[i][1] > arr[j][1]:
      rank[j] += 1
      
print(*rank)

File: 정건우/7주차/260210.java
//https://www.acmicpc.net/problem/20162
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class BOJ_S2_20162_간식파티 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int N = Integer.parseInt(br.readLine());
        int [] arr = new int[N];
        int [] dp = new int[N];

        for (int i = 0; i < N; i++) {
            int score = Integer.parseInt(br.readLine());
            arr[i] = score;
            dp[i] = score;
        }

        for (int i = 1; i < N; i++) {
            for (int j = 0; j < i; j++) {
                if (arr[j] >= arr[i]) continue;

                dp[i] = Math.max(dp[i], dp[j]+arr[i]);
            }
        }

        int max = 0;
        for (int i = 0; i < N; i++) {
            max = Math.max(max, dp[i]);
        }

        System.out.println(max);

    }
}
